modules/signals/spec/select-signal.spec.ts

```diff
@@ -0,0 +1,208 @@
+import { effect, isSignal, signal } from '@angular/core';
+import { selectSignal } from '../src';
+import { testEffects } from './helpers';
+
+describe('selectSignal', () => {
```

Comment:
I didn't look the PR yet, but what's your opinion of adding an integration test from the beginning, and build on-top of it in future PRs?

Reply:
There are already some kind of integration tests in `signal-state.spec.ts` that test the behavior of `signalState` that uses `toDeepSignal` and `selectSignal` under the hood. But yes, good idea! I'll add more in upcoming PRs.